app/models/booking.rb

```diff
@@ -0,0 +1,21 @@
+class Booking < ApplicationRecord
+  belongs_to :user
+  belongs_to :listing
+
+  STATUS = ["Accepted", "Pending", "Cancelled"]
+
+  validates :start_date, presence: true
+  validates :end_date, presence: true
+  validate :end_date_is_after_start_date
+  validates :status, presence: true, inclusion: { in: STATUS, message: "%{value} is not a valid status" }
+
+  private
+
+  def end_date_is_after_start_date
+    return if end_date.blank? || start_date.blank?
+
+    if end_date < start_date
+      errors.add(:end_date, "cannot be before the start date") 
```

Comment:
Should this be "errors" or "error"?

Reply:
It is `errors` for [this ](https://devdocs.io/rails~6.1/activemodel/errors#method-i-add)class method of active model.